.github/workflows/build.yml

```diff
@@ -8,31 +8,54 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ["12.x", "14.x", "16.x"]
     steps:
       - name: Clone Repository
         uses: actions/checkout@v2
         with:
-          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
-      - name: Use Node.js 12.x
+          fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Use Node.js ${{ matrix.node }}
         uses: actions/setup-node@v2
         with:
-          node-version: '12.x'
+          node-version: ${{ matrix.node }}
       - name: Install dependencies
         run: npm ci
       - name: Build
         run: npm run build:prod:gh-pages --if-present
       - name: Test CI
         run: npm run test:ci
-      - name: SonarCloud Scan
-        uses: SonarSource/sonarcloud-github-action@master
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
       - name: Upload dist/MobProgramming folder
         uses: actions/upload-artifact@v2
         with:
           name: mob-programming-app
           path: dist/MobProgramming
+      - name: Cache coverage folder
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-coverage
+        with:
+          path: coverage
+          key: ${{ runner.os }}-build-${{ env.cache-name }}
+  sonar:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone Repository
+        uses: actions/checkout@v2
+      - name: Cache coverage folder
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-coverage
+        with:
+          path: coverage
+          key: ${{ runner.os }}-build-${{ env.cache-name }}
```

Comment:
I'm not sure about this step which looks exactly like the `build` job (I don't know about `actions/cache@v2` usage).
We can talk about it IRL, then we'll reply to this comment to provide an answer. 😉

Reply:
`actions/cache@v2` allows you to share data between different jobs of the same (or different) workflows using the `key` parameter as reference. A cache step per job is needed to use the existing cache.
In hindsight `artifacts` are a better fit for this usecase, as existing cached content can't be updated without changing the key : [doc](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action)